datageneration/generate_combination_table.py

```diff
@@ -101,21 +106,20 @@ def generate_entities(self, max_number_of_entities_in_prompt: int, max_number_of
                 continue
             selected_entity_numbers.append(selected_idx_for_combinations)
             selected_tag_comb = self.entity_tag_combinations[selected_idx_for_combinations]
-            associated_descriptors = selected_tag_comb['descriptors']
+            associated_descriptors = selected_tag_comb.descriptors
 
             if "brand name" in associated_descriptors:
                 brand_examples = self.property_generator.select_named_property_example("brand~***example***")
                 entity_name = "brand: " + np.random.choice(brand_examples)
             else:
                 entity_name = np.random.choice(associated_descriptors)
+            is_area = selected_tag_comb.is_area
 
             # Randomise whether probabilities should be added to ensure high enough ratio of zero property cases
             add_properties = np.random.choice([True, False], p=[percentage_of_entities_with_props,
                                                                 1 - percentage_of_entities_with_props])
             if add_properties and max_number_of_props_in_entity >= 1:
-                candidate_properties = selected_tag_comb['tag_properties']
-                candidate_properties = list(map(lambda candidate_property: TagProperty(**candidate_property),
-                                                candidate_properties))
+                candidate_properties = selected_tag_comb.tag_properties
```

Comment:
if one entity does not have candidate properties, we skip it. But we might end up insufficient entities with properties. Perhaps we should use while loop to force selection of entities with properties.